buildSrc/src/main/java/quilt/internal/tasks/mappings/EnigmaMappingsServerTask.java

```diff
@@ -0,0 +1,70 @@
+package quilt.internal.tasks.mappings;
+
+import org.gradle.api.Project;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.JavaExec;
+import quilt.internal.Constants;
+import quilt.internal.tasks.MappingsTask;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class EnigmaMappingsServerTask extends JavaExec implements MappingsTask {
+	@InputFile
+	private final RegularFileProperty jarToMap;
+	@InputFile
+	private final RegularFileProperty mappings;
+
+	private final List<String> serverArgs;
+
+	public EnigmaMappingsServerTask() {
+		Project project = this.getProject();
+		this.setGroup(Constants.Groups.MAPPINGS_GROUP);
+		this.getMainClass().set("org.quiltmc.enigma.network.DedicatedEnigmaServer");
+		this.classpath(project.getConfigurations().getByName("enigmaRuntime"));
+		this.jvmArgs("-Xmx2048m");
+
+		this.jarToMap = getObjectFactory().fileProperty();
+		this.mappings = getObjectFactory().fileProperty();
+		this.mappings.convention(() -> this.getProject().file("mappings"));
```

Comment:
```suggestion
		this.getMappings().convention(project.getLayout().dir(project.provider(() -> project.file("mappings"))));
```
oops forgot to add this one
not sure if there's a cleaner way to get a `Provider<Directory>`